File: src/lc_0070.rs
// climbing stairs
pub struct Solution;

impl Solution {
    // basically just fibonacci
    pub fn climb_stairs(n: i32) -> i32 {
        let mut a = 0;
        let mut b = 1;

        for _ in 2..=n {
            let temp = a + b;
            a = b;
            b = temp;
        }
        b
    }
}

#[cfg(test)]
mod test_0070 {
    use super::*;

    #[test]
    fn eg() {
        println!("{}", Solution::climb_stairs(2));
    }
}

File: leetcode-go/problems/dynamic1D/lc_0070.go
package dynamic1d

// Climbing Stairs

func climbStairs(n int) int {
	n++
	fib := make([]int, n+1)
	fib[0], fib[1] = 0, 1
	for i := 2; i <= n; i++ {
		fib[i] = fib[i-1] + fib[i-2]
	}
	return fib[n]
}
